sdk/resourcemanager/Azure.ResourceManager.Core/src/GenericResourceOperations.cs

```diff
@@ -76,7 +78,7 @@ public async Task<ArmResponse<Response>> DeleteAsync(CancellationToken cancellat
         /// </remarks>
         public ArmOperation<Response> StartDelete(CancellationToken cancellationToken = default)
         {
-            _apiVersion ??= ClientOptions.ApiVersions.LoadApiVersion(ProviderOperations, Id, cancellationToken);
+            _apiVersion = ClientOptions.ApiVersions.TyrGetApiVersion(ProviderOperations, Id, cancellationToken);
```

Comment:
I still think we should use the ??= here unless I am missing something?

Reply:
What if the API version is update from time of construction to time of this call? I would think we want to use the one that was set post construction not what the apiVersion was at construction or on previouse call.

For that matter, the _apiVersion variable is not even needed